redash/query_runner/hive_ds.py

```diff
@@ -97,7 +114,42 @@ def run_query(self, query, user):
 
         connection = None
         try:
-            connection = hive.connect(**self.configuration.to_dict())
+            if self.configuration.get('use_http', False):
+                # default to https
+                scheme = self.configuration.get('http_scheme', 'https')
+
+                # if path is set but is missing initial slash, append it
+                path = self.configuration.get('http_path', '')
+                if path and path[0] != '/':
+                    path = '/' + path
+
+                # if port is set prepend colon
+                port = self.configuration.get('port', '')
+                if port:
+                    port = ':' + port
+
+                # build uri
+                http_uri = "{}://{}{}{}".format(scheme, host, port, path)
```

Comment:
Why build the URL and not take a URL in the configuration?

Reply:
I'm not sure what you mean. The http_uri is used to create the transport on line 135.

Reply:
If your question is: why not make the URL the input fields instead of each separate piece (i.e. host, port) as an input field, the answer is that having each piece of the string makes it easier to use each piece elsewhere if needed.